1-js/99-js-misc/03-currying-partials/article.md

```diff
@@ -162,35 +162,35 @@ function curried(...args) {
+1. Panggilan pertama `curried(1)` mengingat `1` di dalam lingkungan leksikal-nya, dan mengembalikan pembungkus `pass`.
+2. Pembungkus `pass` dipanggil dengan `(2)`: ia menambil argumen sebelumnya (`1`), menggabungkannya `(2)` kemudian memanggil `curried(1, 2)` secara bersama-sama. Karena jumlah argumen masih kurang dari 3, `curry` mengembalikan `pass`.
+3. Pembungkus `pass` dipanggil lagi dengan `(3)`,  untuk pemanggilan berikutnya `pass(3)` mengambil argumen sebelumnya (`1`, `2`) dan menambahkan `3`, membuat panggilan `curried(1, 2, 3)` -- terdapat argumen `3` pada akhirnya, kemudian mereka akan diberikan ke fungsi aslinya.
 
-If that's still not obvious, just trace the calls sequence in your mind or on paper.
+Jika masih belum jelas, cukup lacak lagi urutan pemanggilan dalam benak anda atau coba tulis di kertas.
 
 ```smart header="Fixed-length functions only"
-The currying requires the function to have a fixed number of arguments.
+Currying membutuhkan fungsi untuk memiliki sejumlah argumen tetap.
 
-A function that uses rest parameters, such as `f(...args)`, can't be curried this way.
+Fungsi yang menggunakan sisa parameter, seperti `f(...args)`, tidak bisa di currying dengan cara ini.
 ```
 
 ```smart header="A little more than currying"
-By definition, currying should convert `sum(a, b, c)` into `sum(a)(b)(c)`.
+Menurut definisi, currying harus mengubah `sum(a, b, c)` menjadi `sum(a)(b)(c)`.
 
-But most implementations of currying in JavaScript are advanced, as described: they also keep the function callable in the multi-argument variant.
+Namun sebagian besar implementasi currying di JavaScript bersifat lanjutan, ssperti yang dijelaskan: implementasi tersebut juga membuat fungsi dapat dipanggil dalam bentuk multi-argumen.
 ```
 
-## Summary
+## Ringkasan
 
-*Currying* is a transform that makes `f(a,b,c)` callable as `f(a)(b)(c)`. JavaScript implementations usually both keep the function callable normally and return the partial if the arguments count is not enough.
+*Currying* adalah transformasi yang membuat `f(a,b,c)` dapat dipanggil sebagai `f(a)(b)(c)`. Implementasi JavaScript biasanya membuat fungsi dapat dipanggil secara normal dan mengembalikan dalam bentuk parsial jika jumlah argumen tidak cukup.
 
-Currying allows us to easily get partials. As we've seen in the logging example, after currying the three argument universal function `log(date, importance, message)` gives us partials when called with one argument (like `log(date)`) or two arguments (like `log(date, importance)`).  
+_Currying_ memungkinkan kita untuk mendapatkan sebuah bagian. Seperti yang kita lihat di contoh logging, setelah _currying_ tiga argumen dari fungsi universal `log(date, importance, message)` akan memberikan kita fungsi parsial ketika dipanggil dengan satu argumen (seperti `log(date)`) atau dua argumen (seperti `log(date, importance)`).  
```

Comment:
I'm not sure with this

Reply:
kwkw, agak kurang sreg ya?
Ini bukan materi buat yang newbie sih keknya, dah mayan tepat kayanya.
partials = sebagian.

Reply:
wkw iya